### ISSUE TITLE
Action is using deprecated "set-output" command

### ISSUE THREAD

**Issue by mblomdahl** [tchupp/actions-terraform-pr#25]
Hello @tchupp & @ebachle! Thanks for a good plugin for posting the Terraform state changes. I started using it today, but as you see in the logs here, https://github.com/mblomdahl/tfc-vs4345-k8s/actions/runs/3796399942/jobs/6456457736, I keep getting this deprecation warning about its use of `set-output`:

> Warning: The `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

If you guys would have some time/effort to address this, I would be very happy! Or, if you want some help/collaboration on resolving it, I can maybe try helping out.

**Comment by ebachle** [comment]
Hey @mblomdahl, both @tchupp and I are aware of that deprecation and it's something we've been meaning to look at.

I can actually take a look at potentially doing it today and see if it works.  I'll double check it against one of our larger projects, because we've run into max size issues with envvars (you can see it in how we transfer the plan outputs via file here: https://github.com/tchupp/actions-terraform-pr/blob/main/action.yml#L118

So that's the one thing I'll check before converting it over.  It should be a pretty easy update if that's not an issue.

**Comment by ebachle** [comment]
I'm going to do this in two steps, because one is us changing our code, and the other is us having to update a dependency.  

The first PR #26 is going to fix ours and that tested cleanly.  

The other is going to be to update to the most recent copy of https://github.com/actions/github-script, because in a recent patch, the maintainers updated that.  https://github.com/actions/github-script/releases/tag/v6.3.2.  But that's two major version hops, so I want to test that separately.

**Comment by ebachle** [comment]
@mblomdahl this should all be resolved in https://github.com/tchupp/actions-terraform-pr/releases/tag/v1.0.22, or if you're using just the v1 tag, it should be set there as well!  

Thanks for brining it up!

**Comment by mblomdahl** [comment]
@ebachle Thanks for looking into this so swiftly! I tried pulling in the new changes and verifying in my project, but it does not seem that the release is fully published...? I noticed an error in this "Release Tagger" workflow run here, https://github.com/tchupp/actions-terraform-pr/actions/runs/3796828973/jobs/6457272468

<img width="1265" alt="image" src="https://user-images.githubusercontent.com/786326/209882573-39d95699-b431-480a-b9a5-dfee9b012f93.png">

When I ran my first PR with "`@v1`", then I noticed that many deprecation warnings were still there as if nothing had been updated (see GHA output from mblomdahl/tfc-vs4345-k8s#12). Then, when I updated my other PR to use "`@v1.0.22`" specifically in mblomdahl/tfc-vs4345-k8s#11, I get the following error message:

> **Error:** Unable to resolve action `hashicorp/setup-terraform@v1.0.22`, unable to find version `v1.0.22`

Does this make any sense to you?

**Comment by mblomdahl** [comment]
Sorry, false alarm. It was "skit bakom spakarna" as we say in 🇸🇪... I added "`@v1.0.22`" to the wrong GHA dependency. Now it resolves! I added #28 however to track the publishing issue.

I think there might be a few less `set-output` deprecation warnings in the end, but at the early stages of [the run](https://github.com/mblomdahl/tfc-vs4345-k8s/actions/runs/3797202443/jobs/6457933707), we still have a bunch to go! 🙃 

<img width="845" alt="image" src="https://user-images.githubusercontent.com/786326/209886323-39bfcbf4-a439-47bd-a533-c714005ac675.png">

**Comment by ebachle** [comment]
Heya @mblomdahl, I did a bit of digging and I think I know what's going on there and what the difference is between the test pipeline I ran it on and yours.  

What it boils down to is the version of `hashicorp/setup-terraform` that you're using higher up your pipeline: 

The reason I needed to upgrade the `actions/github-script` to v5+ in one of my PRs is because it update the version of the NPM package `@actions/core` to `1.10.0`.  

In the v1 line of the `setup-terraform` they're still using @actions/core that's a bit older.  https://github.com/hashicorp/setup-terraform/blob/v1/package.json#L22

If you bump to the v2 version of `hashicorp/setup-terraform`, then that should clean up as it's ultimately what's hiding behind that `terraform-bin` wrapper that's being executed in your pipeline.  

I'll make sure that our examples for using this though don't reference v1, so that nobody picks it up from us.
